modules/index-service/src/main/java/org/opencastproject/index/service/message/WorkflowMessageReceiverImpl.java

```diff
@@ -71,110 +49,69 @@ public WorkflowMessageReceiverImpl() {
-                  ExceptionUtils.getStackTrace(e));
-        }
-        return;
-      case AddDefinition:
-        // TODO: Update the index with it as soon as the definition are part of it
-        return;
-      case DeleteDefinition:
-        // TODO: Update the index with it as soon as the definition are part of it
-        return;
+        deleteWorkflowInstance(workflowItem);
+        break;
       default:
         throw new IllegalArgumentException("Unhandled type of WorkflowItem");
     }
   }
 
-  /** OSGi DI. */
-  public void setWorkspace(Workspace workspace) {
-    this.workspace = workspace;
-  }
-
-  /** OSGi DI. */
-  public void setAuthorizationService(AuthorizationService authorizationService) {
-    this.authorizationService = authorizationService;
+  private void deleteWorkflowInstance(WorkflowItem workflowItem) {
+    final String organization = getSecurityService().getOrganization().getId();
+    final User user = getSecurityService().getUser();
+    final String eventId = workflowItem.getId();
+    logger.debug("Received Delete Workflow instance Entry {}", eventId);
+
+    // Remove the Workflow instance entry from the search index
+    try {
+      getSearchIndex().deleteWorkflow(organization, user, eventId, workflowItem.getWorkflowInstanceId());
+      logger.debug("Workflow instance media package {} removed from search index", eventId);
+    } catch (NotFoundException e) {
+      logger.warn("Workflow instance media package {} not found for deletion", eventId);
+    } catch (SearchIndexException e) {
+      logger.error("Error deleting the Workflow instance entry {} from the search index", eventId, e);
+    }
   }
 
-  /** OSGi callback for acl services. */
-  public void setAclServiceFactory(AclServiceFactory aclServiceFactory) {
-    this.aclServiceFactory = aclServiceFactory;
+  private void updateWorkflowInstance(WorkflowItem workflowItem) {
+    logger.debug("Received Update Workflow instance Entry for index {}", getSearchIndex().getIndexName());
+    final String organization = getSecurityService().getOrganization().getId();
+    final User user = getSecurityService().getUser();
+    final String eventId = workflowItem.getId();
+    final MediaPackage mediaPackage = workflowItem.getMediaPackage();
+
+    // Load or create the corresponding recording event
+    try {
+      Event event = getOrCreateEvent(eventId, organization, user, getSearchIndex());
+      event.setCreator(user.getName());
+      event.setWorkflowId(workflowItem.getWorkflowInstanceId());
+      event.setWorkflowDefinitionId(workflowItem.getWorkflowDefinitionId());
+      event.setWorkflowState(workflowItem.getState());
+      event.setAccessPolicy(workflowItem.getAccessControlListJSON());
+
+      // Update metadata
+      DublinCoreCatalog dcCatalog = workflowItem.getEpisodeDublincoreCatalog();
+      if (dcCatalog != null) {
+        updateEvent(event, dcCatalog);
+      }
+      event.setSeriesName(mediaPackage.getSeriesTitle());
```

Comment:
updateEvent on line 106 also does this (see https://github.com/opencast/opencast/blob/develop/modules/index-service/src/main/java/org/opencastproject/index/service/impl/index/event/EventIndexUtils.java#L506) , so line 103 can be removed.

Reply:
Makes sense → updated.

Reply:
Good catch → fixed